lib/core/src/connection_pool.cpp

```diff
@@ -279,7 +351,9 @@ namespace irods
             if (lock.try_lock()) {
                 if (!conn_ctxs_[i].in_use.load()) {
                     conn_ctxs_[i].in_use.store(true);
-                    return {*this, *refresh_connection(i), i};
+                    connection_proxy proxy{*this, *refresh_connection(i), i};
+                    ++conn_ctxs_[i].retrieval_count;
```

Comment:
Not quite.

It is done here to keep `refresh_connection` from having multiple responsibilities and it makes it easier for others to see where the counter is bumped.

Reply:
In that case, why not just add `++conn_ctxs_[i].retrieval_count;` before the `return` statement instead of instantiating `proxy`, incrementing, and then `return`ing?